src/messages/mod.rs

```diff
@@ -171,6 +176,28 @@ pub enum DirectMessage {
     ResourceProofResponseReceipt,
     /// Sent from a proxy node to its client to indicate that the client exceeded its rate limit.
     ProxyRateLimitExceeded { ack: Ack },
+    Vote { vote: Vote<NodeState> },
+    Block { block: Block<NodeState> },
+    ChainRequest,
+    ChainResponse {
+        blocks: Vec<Block<NodeState>>,
+        pending_blocks: Vec<Block<NodeState>>,
+    },
+    /// Sent to a newly-joined and relocated node (after passing resource-proofing)
+    NodeApproval {
+        /// The latest valid blocks relating to elders.
+        blocks: Vec<Block<NodeState>>,
+        /// The valid pending blocks relating to elders.
+        pending_blocks: Vec<Block<NodeState>>,
+        // All adults and infants in our section.
+        non_elders: Vec<PublicInfo>,
+        not_yet_valid_blocks: Vec<Block<NodeState>>,
+    },
+    EncryptionKeyRequest { public_sign_key: sign::PublicKey },
+    EncryptionKeyResponse {
+        public_sign_key: sign::PublicKey,
+        public_encrypt_key: Option<box_::PublicKey>,
```

Comment:
I guess this request-response is a point-to-point communication?
so a response with None indicates a failure?